substrate/frame/support/procedural/src/pallet/expand/error.rs

```diff
@@ -102,6 +102,17 @@ pub fn expand_error(def: &mut Def) -> proc_macro2::TokenStream {
 
 	let capture_docs = if cfg!(feature = "no-metadata-docs") { "never" } else { "always" };
 
+	let deprecation = crate::deprecation::get_deprecation_enum(
+		&quote::quote! {#frame_support},
+		&error.attrs,
+		error_item
+			.variants
+			.iter()
+			.enumerate()
+			.map(|(index, item)| (index as u8, item.attrs.as_ref())),
```

Comment:
I don't know if we want to support it but:

User can currently write `#[codec(index = N)]` on variants of error and event. The pallet macro doesn't prevent it AFAIK.

If it happens the `index` here and codec index can be different.

note that scale-info reads the `#[codec(index = N)]` attributes to infer the index.

Reply:
should we check the attribute of the variant then and if `codec.index` is not present then use the definition index?

Reply:
pushed the updated code

Reply:
Great, change looks good to me